subsys/jaxrs/src/main/java/org/commonjava/indy/bind/jaxrs/ApiVersioningFilter.java

```diff
@@ -0,0 +1,317 @@
+ */
+@ApplicationScoped
+public class ApiVersioningFilter
+                implements Filter
+{
+
+    private final Logger logger = LoggerFactory.getLogger( getClass() );
+
+    @Inject
+    private IndyVersioning indyVersioning;
+
+    private List<DeprecatedApiEntry> deprecatedApis = new ArrayList<>();
+
+    private String minApiVersion;
+
+    public ApiVersioningFilter()
+    {
+    }
+
+    public ApiVersioningFilter( IndyVersioning versioning )
+    {
+        this.indyVersioning = versioning;
+    }
+
+    @Override
+    public void init( final FilterConfig filterConfig ) throws ServletException
+    {
+        parseDeprecatedApis();
+    }
+
+    private void parseDeprecatedApis()
+    {
+        Float minVersion = 0f;
+
+        Properties props = indyVersioning.getDeprecatedApis();
+        Set<String> keys = props.stringPropertyNames();
+        for ( String key : keys )
+        {
+            DeprecatedApiEntry et;
+            String value = props.getProperty( key );
+
+            Float startVersion;
+            Float endVersion;
+
+            if ( key.indexOf( "," ) >= 0 ) // range
+            {
+                key = key.replaceAll( "[\\[|\\]]", "" ); // strip off square brackets if present
+                String[] kv = key.split( "," );
+                startVersion = Float.parseFloat( kv[0].trim() );
+                endVersion = Float.parseFloat( kv[1].trim() );
+                et = new DeprecatedApiEntry( new FloatRange( startVersion, endVersion ), value);
+            }
+            else
+            {
+                endVersion = Float.parseFloat( key.trim() );
+                et = new DeprecatedApiEntry( endVersion, value );
+            }
+
+            // Calculate minApiVersion
+            if ( et.isOff() )
+            {
+                minVersion = endVersion + 0.1f;
+            }
+            deprecatedApis.add( et );
+        }
+
+        minApiVersion = minVersion.toString();
+
+        logger.debug( "Parsed deprecatedApis:{}, minApiVersion:{}", deprecatedApis, minApiVersion );
+    }
+
+    @Override
+    public void doFilter( final ServletRequest request, final ServletResponse response, final FilterChain chain )
+                    throws IOException, ServletException
+    {
+        HttpServletRequest httpServletRequest = (HttpServletRequest) request;
+        HttpServletResponse httpServletResponse = (HttpServletResponse) response;
+
+        String reqApiVersion = httpServletRequest.getHeader( HEADER_INDY_API_VERSION );
+
+        // Inject header 'Indy-API-Version' and 'Indy-Min-API-Version' into outgoing responses
+        String respApiVersion = reqApiVersion;
+        if ( isBlank( respApiVersion ) )
+        {
+            respApiVersion = indyVersioning.getApiVersion(); // respond with default version
+        }
+        httpServletResponse.addHeader( HEADER_INDY_API_VERSION, respApiVersion );
+        httpServletResponse.addHeader( HEADER_INDY_MIN_API_VERSION, minApiVersion );
```

Comment:
Let's add one that gives the current API version too...since we're returning the one sent by the user.

Reply:
Ok